make/test.mk

```diff
@@ -250,3 +274,33 @@ ifeq ($(IS_OS_3),)
 else
 	sed -e 's|REPLACE_IMAGE|${IMAGE_NAME}|g' ${E2E_REPO_PATH}/deploy/operator.yaml | oc apply -f -
 endif
+
+.PHONY: build-and-deploy-service
+build-and-deploy-service:
+# build service
+# when e2e tests are triggered from different repo - eg. as part of PR in host-operator repo - and the image of the deployment is (not) provided
+ifeq ($(SET_IMAGE_NAME),)
+	# check if it is running locally
+    ifeq ($(OPENSHIFT_BUILD_NAMESPACE),)
+        ifneq ($(IS_OS_3),)
+			# using OS 3 (assuming it's minishift) then build image
+			$(eval IMAGE_NAME := quay.io/${GO_PACKAGE_ORG_NAME}/${REPO_NAME}:${DATE_SUFFIX})
+			$(MAKE) -C ${REPO_PATH} build
+			docker build -f ${REPO_PATH}/build/Dockerfile -t ${IMAGE_NAME} ${REPO_PATH}
+        else
+			# using OS 4 then build and push image
+			$(eval IMAGE_NAME := quay.io/${QUAY_NAMESPACE}/${REPO_NAME}:${DATE_SUFFIX})
+			$(MAKE) -C ${REPO_PATH} build
+			docker build -f ${REPO_PATH}/build/Dockerfile -t ${IMAGE_NAME} ${REPO_PATH}
+			docker push ${IMAGE_NAME}
+        endif
+    else
+		# if it is running in CI than we expect that it's PR for toolchain-e2e repo (none of the images was provided), so use name that was used by openshift-ci
+		$(eval IMAGE_NAME := registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:${REPO_NAME})
```

Comment:
Can you please make sure that we are building this image in openshift-ci? I don't remember we have this image available in openshift-ci image registry

Reply:
@dipak-pawar are you talking about this thing ?? https://github.com/openshift/release/blob/master/ci-operator/config/codeready-toolchain/toolchain-e2e/codeready-toolchain-toolchain-e2e-master.yaml#L16-L32

Can you plz provide me some doc link to do the thing you asking for ??

Reply:
Done https://github.com/openshift/release/pull/5378